libs/vkd3d/resource.c

```diff
@@ -1543,17 +1544,26 @@ static void STDMETHODCALLTYPE d3d12_resource_Unmap(d3d12_resource_iface *iface,
 static D3D12_RESOURCE_DESC * STDMETHODCALLTYPE d3d12_resource_GetDesc(d3d12_resource_iface *iface,
         D3D12_RESOURCE_DESC *resource_desc)
 {
-    struct d3d12_resource *resource = impl_from_ID3D12Resource1(iface);
+    struct d3d12_resource *resource = impl_from_ID3D12Resource2(iface);
 
     TRACE("iface %p, resource_desc %p.\n", iface, resource_desc);
 
-    *resource_desc = resource->desc;
+    resource_desc->Dimension = resource->desc.Dimension;
```

Comment:
Not related to the patch, but I noticed that this function signature looks completely different from https://docs.microsoft.com/en-us/windows/win32/api/d3d12/nf-d3d12-id3d12resource-getdesc. Shouldn't we return the struct by value?

Reply:
This signature is correct, any function that returns a struct is turned into a function that takes a pointer to an out-struct and returns the same pointer. Don't ask me why, but this is done for all COM functions of this kind.

Reply:
OK.